src/components/Button/Button.tsx

```diff
@@ -1,37 +1,68 @@
+import "./Button.scss";
 import { ReactNode } from "react";
 import cx from "classnames";
-import "./Button.scss";
+import ButtonLink from "./ButtonLink";
+
+type ButtonVariant = "primary" | "primary-action" | "semi-clear" | "clear";
 
-type Props = {
+type ButtonProps = React.HTMLProps<HTMLButtonElement> & {
```

Comment:
mm if you use React.HTMLProps<HTMLButtonElement>, logically it's need to pass all the props: <button {...rest}/> otherwise it's not obvious why some prop will not used by the button but existing in type ButtonProps